File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
    const n = moves.length;
    const stack = [];
    let answer = 0;
    for(let i = 0; i < n; i++){
        const place = moves[i] - 1;
        for(let j = 0; j < board.length; j++){
            if(board[j][place] !== 0){
                const item = board[j][place];
                if(stack.length > 0 && stack[stack.length - 1] === item){
                    stack.pop();
                    answer += 2;
                } else {
                    stack.push(item);
                }
                board[j][place] = 0;
                break;
            }
        }
    }
    console.log(stack, board)
    return answer;
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    n = len(board)
    answer = 0
    stack = []
    for c in moves:
        for row in board:
            if row[c - 1] != 0:
                if stack and stack[len(stack) - 1] == row[c - 1]:
                    stack.pop()
                    answer += 2
                else:
                    stack.append(row[c - 1])
                row[c - 1] = 0
                break
    print(board)
    return answer